### PR TITLE
Removed Globals from BibDatabaseContext

### PR DESCRIPTION
**Pull request by oscargus**
This almost worked. The penalty was that I had to (temporarily) introduce it in `FileUtil.expandFilename` where the arbitrary extension line makes it extremely cumbersome to get around it.

### CODE REVIEW

File: src/main/java/net/sf/jabref/external/WriteXMPAction.java [hunk 1]
```diff
@@ -121,11 +121,16 @@ public void run() {
             List<File> files = new ArrayList<>();
 
             // First check the (legacy) "pdf" field:
-            entry.getFieldOptional(FieldName.PDF).ifPresent(pdf ->
-                FileUtil.expandFilename(pdf, panel.getBibDatabaseContext().getFileDirectory("pdf"))
-                    .ifPresent(files::add));
+            entry.getFieldOptional(FieldName.PDF)
+                    .ifPresent(
+                            pdf -> FileUtil
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/external/WriteXMPAction.java]: The formatting here and below looks a bit weird...

File: src/main/java/net/sf/jabref/external/WriteXMPEntryEditorAction.java [hunk 2]
```diff
@@ -55,11 +55,16 @@ public void actionPerformed(ActionEvent actionEvent) {
         List<File> files = new ArrayList<>();
 
         // First check the (legacy) "pdf" field:
-        entry.getFieldOptional(FieldName.PDF).ifPresent(pdf -> FileUtil
-                .expandFilename(pdf, panel.getBibDatabaseContext().getFileDirectory("pdf")).ifPresent(files::add));
+        entry.getFieldOptional(FieldName.PDF)
```
Comment (Siedlerchr) [inline src/main/java/net/sf/jabref/external/WriteXMPEntryEditorAction.java]: see above

Comment (oscargus) [inline src/main/java/net/sf/jabref/external/WriteXMPEntryEditorAction.java]: It does indeed, but that is what Eclipse gives me when reformatting the whole expression...

**Comment by oscargus** [comment]
I think I know how to get around this now. It seems like there are three possible directories "fileDirectory", "pdfDirectory", and "psDirectory", where only the first one is normally used and the other comes from the legacy pdf and ps fields. Now, it might be possible to hard-code in the preferences support for more extensions, but do we really need to support that or would it be OK to include a limitation to only those three?

**Comment by Siedlerchr** [comment]
LGTM, jut two weird formatting things.
Regardng the FileUtil class. I will rework this., because it contains a lot of outdated code which can be replaced with Standard Java implementation. Maybe we could get rid of some prefs there.

Regarding the directories, there is also a discussion about the import/export dir preferences #1813 
I don't think we need the legacy pdf/ps fields. Isn't there a cleanup job to move them?

**Comment by oscargus** [comment]
There was an old issue where someone used them for something completely different, so maybe we shouldn't remove them. As long as it is only those three (or at least a well known subset of all possible `extDirectory` it should be OK. Now the problem is that one can use any extension and then the code tries to find a dedicated directory for that extension, which cannot be generated in any other way than editing the preference XML...

**Comment by oscargus** [comment]
#496 I think. Although that doesn't discuss arbitrary extensions (from a quick browse).

**Comment by lenhard** [comment]
I would also say that it is ok to remove support for the legacy file directories. And I do not really see why we should support arbitrary extension directories.

**Comment by oscargus** [comment]
OK, dropping arbitrary extensions I'm fine with, but I will not remove support for the pdf and ps directories right now.

File: src/main/java/net/sf/jabref/external/WriteXMPAction.java (hunk 1 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/external/WriteXMPAction.java]: One of them was sorted out with the new code, but not the other...

**Comment by oscargus** [comment]
Now there are no more dependencies on Globals or logic in BibDatabaseContext. Still a dependency on shared though, but that I haven't looked into.

**Pull request merged**